libs/verify-android-environment.sh

```diff
@@ -10,10 +12,22 @@ if [[ ! -f "$(pwd)/libs/build-all.sh" ]]; then
   exit 1
 fi
 
+echo "Verifying desktop-specific environment..."
 # Android consumers are likely to also want to be able to run a quick
 # `cargo build` for their desktop env, so verify that as well.
 "$(pwd)/libs/verify-desktop-environment.sh"
+echo ""
 
+echo "Verifying android-specific environment..."
 "$(pwd)/libs/verify-android-ci-environment.sh"
 
+# Mac-specific checks, to help out with the M1 transition.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  java_arch=$(which java | xargs file | cut -f 5 -d ' ')
+  system_arch=$(uname -m)
+  if [ "$java_arch" != "$system_arch" ]; then
+    echo "WARNING: mismatching Java (${java_arch}) and system (${system_arch}) architectures. Make sure this is intentional, or you may get strange build errors."
```

Comment:
tried running this locally on my Intel machine and got a 
```sh
WARNING: mismatching Java (with
64-bit
64-bit) and system (x86_64) architectures. Make sure this is intentional, or you may get strange build errors.
```

This could be an issue with my own setup though, as I remember I installed multiple versions of java at some point (can't remember 🙃) For reference, here's what `which java | xargs file` prints: 
```sh
/usr/bin/java: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64e:Mach-O 64-bit executable arm64e]
/usr/bin/java (for architecture x86_64):	Mach-O 64-bit executable x86_64
/usr/bin/java (for architecture arm64e):	Mach-O 64-bit executable arm64e
```

If this is a very edge setup (which means I should get my java stuff fixed locally 😅) then nevermind! 😄

Reply:
Oh, interesting. I'll make the arch parsing a bit more robust!